include/fiction/layouts/bounding_box.hpp

```diff
@@ -21,6 +19,20 @@
 
 namespace fiction
 {
+/**
+ * Modes to use for creating the 2D-bounding box.
+ */
+enum class bounding_box_2d_selection
```

Comment:
But first of all, why is it necessary? If the passed layout contains defects, they should be respected. If that is not desired, the defect layer can be "cut off" by `static_cast`ing the layout in the pass to the constructor. See the code example, e.g., [here](https://fiction.readthedocs.io/en/latest/algorithms/path_finding.html#_CPPv4I0000EN7fiction6a_starE4PathRK3LytRK17routing_objectiveI3LytERK16distance_functorI3Lyt4DistERK12cost_functorI3Lyt4CostERK13a_star_params) for such a "down-casting".

Reply:
I agree! That's the way to go! Thanks!